Rename ExtraOptions fields to EncryptionOptions.

rauchenstein opened the pull request:
In practice, this field only encodes encryption keys.
There are several places in the code base where this field being
non-empty is used as an indicator that encryption is enabled, so using
it for other purposes at this point could easily lead to bugs.

Fixes #62358.

rauchenstein committed:
rename ExtraOptions fields to EncryptionOptions

In practice, this field only encodes encryption keys.
There are several places in the code base where this field being
non-empty is used as an indicator that encryption is enabled, so using
it for other purposes at this point could easily lead to bugs.

Fixes #62358.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63864)
<!-- Reviewable:end -->

itsbilal reviewed (commented):
Great work so far! Just some minor, comments-related comments (hah).

We usually leave a release note for every change / PR. Since there's no user-facing change here, you should end the commit message and PR description with `Release note: None.`. Without it, I think the git hooks will complain.

Also, one thing the welcome doc mentioned but the issue didn't, was that it'd also be good to the flow of a StoreSpec to a StorageConfig. This could be done in the form of a comment on StorageConfig saying where/how it is created from a StoreSpec. StoreSpec already has a comment mentioning that it comes from the `--store` flag.

> Reviewed 10 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63864#-:-MYfrTvU1vOuA23gfoU5:brlr34d)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pbardea and @rauchenstein)

___
*[pkg/base/config.go, line 511 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/63864#-MYfrRMj9g2BQ8ye0nKs:-MYfrRMk91az54J-gaxz:bn1lsb5) ([raw file](https://github.com/cockroachdb/cockroach/blob/0b486fef523736b2ab10fb6ee9b434cca4fa1931/pkg/base/config.go#L511)):*
> ```Go
> 	UseFileRegistry bool
> 	// EncryptionOptions is a serialized protobuf set by Go CCL code and passed through
> 	// to C CCL code.
> ```

The comment here could also mention two things:

1) the presence of this field means encryption is enabled, and
2) these options are used in CCL code to set up encryption-at-rest.
___
*[pkg/base/store_spec.go, line 186 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/63864#-MYfsf7-1i8h9ONOlodD:-MYfsf70-UcJeWFw2-Xz:b-fihtsw) ([raw file](https://github.com/cockroachdb/cockroach/blob/0b486fef523736b2ab10fb6ee9b434cca4fa1931/pkg/base/store_spec.go#L186)):*
> ```Go
> 	PebbleOptions string
> 	// EncryptionOptions is a serialized protobuf set by Go CCL code and passed
> 	// through to C CCL code.
> ```

Likewise


<!-- Sent from Reviewable.io -->

rauchenstein committed:
add IsEncrypted() method to StoreSpec and StorageConfig to serve as source-of-truth

Previously this determination was made by testing whether the EncryptionOptions
field was empty.  This field is an encoded protobuf and is otherwise treated
as opaque, so interpreting it this way is fragile.

Initial implementation of IsEncrypted() simply factors out the existing
behavior, but going forward this method serves as source-of-truth if that
behavior needs to change (for example, if EncryptionOptions has reason
to carry some data without encryption-at-rest being fully enabled).

Release note: None

itsbilal reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> great work! I like the way it worked out in the second commit too.

> Reviewed 6 of 6 files at r2, 5 of 5 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63864#-:-MZUBw3H2FwIJZrjgJr7:b-im00do)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rauchenstein)



<!-- Sent from Reviewable.io -->

rauchenstein commented:
bors r+
